modules/rubiconBidAdapter.js

```diff
@@ -1305,4 +1186,70 @@ export function resetUserSync() {
   hasSynced = false;
 }
 
+function setBidFloors(bidRequest, imp) {
+  if (imp.bidfloorcur != 'USD') {
+    delete imp.bidfloor;
+    delete imp.bidfloorcur;
+  }
+
+  let bidFloor = parseFloat(deepAccess(bidRequest, 'params.floor'));
+
+  if (!isNaN(bidFloor)) {
+    imp.bidfloor = bidFloor;
+    imp.bidfloorcur = 'USD';
+  }
+}
+
+function addMultibid(data, multibid) {
+  if (!multibid) {
+    return;
+  }
+  deepSetValue(data, 'ext.prebid.multibid', multibid.reduce((result, i) => {
+    let obj = {};
+
+    Object.keys(i).forEach(key => {
+      obj[key.toLowerCase()] = i[key];
+    });
+
+    result.push(obj);
+
+    return result;
+  }, []));
+}
+
+function addOrtbFirstPartyData(data, nonBannerRequests) {
+  let fpd = {};
+  const keywords = new Set();
+  nonBannerRequests.forEach(bidRequest => {
+    const bidFirstPartyData = {
+      user: {ext: {data: {...bidRequest.params.visitor}}},
+      site: {ext: {data: {...bidRequest.params.inventory}}}
+    };
+
+    // add site.content.language
+    const impThatHasVideoLanguage = data.imp.find(imp => imp.ext?.prebid?.bidder?.rubicon?.video?.language);
+    if (impThatHasVideoLanguage) {
+      bidFirstPartyData.site.content = {
+        language: impThatHasVideoLanguage.ext?.prebid?.bidder?.rubicon?.video?.language
+      }
+    }
+
+    if (bidRequest.params.keywords) {
+      const keywordsArray = (!Array.isArray(bidRequest.params.keywords) ? bidRequest.params.keywords.split(',') : bidRequest.params.keywords);
+      keywordsArray.forEach(keyword => keywords.add(keyword));
+    }
+    fpd = mergeDeep(fpd, bidRequest.ortb2 || {}, bidFirstPartyData);
+
+    // add user.id from config.
+    // NOTE: This is DEPRECATED. user.id should come from setConfig({ortb2}).
+    const configUserId = config.getConfig('user.id');
+    fpd.user.id = fpd.user.id || configUserId;
+  });
+
+  mergeDeep(data, fpd);
+
+  deepSetValue(data, 'site.keywords', Array.from(keywords.values()).join(','));
+  delete data?.ext?.prebid?.storedrequest;
```

Comment:
I know we did this before, but perhaps there are use cases where we do want to pass storedRequests to PBS? For now I guess we should leave it but we should think about setting up a way to pass them eventually

Reply:
This question should be passed to @bretg ; for now all I can say is that on 10/11/2022 Bret answered this way (https://magnite.atlassian.net/browse/HB-15147) regarding storedrequest: 

 2. Another test fails because it DOESN’T expect to find data.imp[0].ext.prebid.storedrequest and data.ext.prebid.storedrequest. Indeed, the library adds it. Should I remove the storedrequests or the tests ?

The existing code covers this situation. We need to remove any storedrequest specified.

     // set storedrequests to undefined so not sent to PBS

      // top level and imp level both 'ext.prebid' objects are set above so no exception thrown here

      data.ext.prebid.storedrequest = undefined;

      data.imp[0].ext.prebid.storedrequest = undefined;

Reply:
Ok well let's just leave it out and delete it for now.

Does the code delete both top level and imp level SR's?